src/transaction/deserialize.c

```diff
@@ -10,6 +10,18 @@
 #include "../common/buffer.h"
 #include "types.h"
 
+bool is_authority_output(uint8_t token_data) {
+    return (token_data & TOKEN_DATA_AUTHORITY_MASK) > 0;
+}
+
+bool is_mint_authority(uint8_t token_data, uint64_t value) {
+    return is_authority_output(token_data) && value == MINT_AUTHORITY_MASK;
```

Comment:
I think this should be `&& (value & MINT_AUTHORITY_MASK) > 0`, shouldn't it?

The same for melt

Reply:
The problem with this is the Mint+Melt case, where an output is authority for both mint and melt.
This edge case may be complicated for the user so I decided not to allow this case in the Ledger app.

Do you think we should allow this case as a special `Mint+Melt output` on the UI?

Reply:
I think it's ok to limit this but I don't see as this complicated, we could show exactly as you wrote: `Mint + Melt` or `Mint and Melt`.

If we decide to leave this out, we should just make clear in the design

Reply:
I forgot to put the reason the discussion was closed, this was meant to be discussed and implemented on another PR.
We can block this PR to solve this issue but this can easily be implemented later depending on the decision.